In [1]:
from pathlib import Path
import pandas as pd

project_root = Path(r"C:\Users\HP-ZBOOK i7\graphrag_test")

pilot1_documents = pd.read_csv(
    project_root / "pilot_01" / "inspection" / "08_documents_all.csv"
)

pilot2_documents = pd.read_csv(
    project_root / "pilot_02" / "inspection" / "08_documents_all.csv"
)

pilot1_text = pilot1_documents.loc[0, "text"]
pilot2_text = pilot2_documents.loc[0, "text"]

print("Pilot 1 documents:", len(pilot1_documents))
print("Pilot 2 documents:", len(pilot2_documents))
print("Pilot 1 characters:", len(pilot1_text))
print("Pilot 2 characters:", len(pilot2_text))
print("Extracted texts identical:", pilot1_text == pilot2_text)

Pilot 1 documents: 1
Pilot 2 documents: 1
Pilot 1 characters: 73134
Pilot 2 characters: 73134
Extracted texts identical: True


## Pilot 2 Document-Conversion Inspection — Conclusion

Pilot 1 and Pilot 2 each processed one source document. The extracted
document text contains 73,134 characters in both pilots, and an exact
string comparison returned `True`.

This demonstrates that the PDF-to-text conversion output was identical
between the two experiments. Consequently, differences in the extracted
entities, relationships, communities, and community reports cannot be
attributed to different document text.

The PDF-conversion limitations previously identified in Pilot 1 remain
present in Pilot 2 because the same converted text was used. These include
page-break markers, irregular spacing, repeated headers and footers, and
line-break hyphenation. However, the Pilot 1 recall audit found that all
30 selected source facts remained present in the text units.

Therefore, PDF conversion is treated as an unchanged experimental control,
not as the principal cause of differences between Pilot 1 and Pilot 2.
The main changed variables are the entity schema and the LLM prompts.